### ISSUE TITLE
[Bug]: [laion-1b] The queryNode SIGABRT: abort with log: 'milvus::SegcoreError' what():  Error:RemoveDir:Directory not empty

### ISSUE THREAD

**Issue by ThreadDao** [milvus-io/milvus#33587]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: cardinal-milvus-io-2.4-deebae70a-20240527
- Deployment mode(standalone or cluster):cluster
- MQ type(rocksmq, pulsar or kafka):  pulsar   
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### argo: [laion1b-test-cron-start-53069600](https://argo-workflows.zilliz.cc/archived-workflows/qa/763e7cd8-592d-4fce-9ef2-d860d95a1b8a?nodeId=laion1b-test-cron-start-53069600) 
1. concurrent: upsert + search + query + load
![image](https://github.com/milvus-io/milvus/assets/27288593/0dcd19f2-ffdb-498b-81bf-dedb5935643b)
 2. queryNode `laion1b-test-2-milvus-querynode-0-dd6f7496f-6sfdr` terminated with log:
 
[qn_6sfdr.log](https://github.com/user-attachments/files/15542405/qn_6sfdr.log)

```
25657 I20240531 17:30:14.165886 181993 index_factory.cc:52] [KNOWHERE][Create][milvus] use key HNSW_fp32 to create knowhere index HNSW with version 4
25658 I20240531 17:30:14.180948   177 DataCodec.cpp:123] [SERVER][DeserializeFileData][milvus] DeserializeFileData_deserialize_duration_ms:106
25659 W20240531 17:30:14.188577 182041 ExceptionTracer.cpp:187] Invalid trace stack for exception of type: milvus::SegcoreError
25660 terminate called after throwing an instance of 'milvus::SegcoreError'
25661   what():  Error:RemoveDir:Directory not empty
25662 SIGABRT: abort
25663 PC=0x7fc92af3d03b m=21361 sigcode=18446744073709551610
25664 signal arrived during cgo execution
25665 
25666 goroutine 61159706 [syscall]:
25667 [2024/05/31 17:30:14.206 +00:00] [INFO] [segments/segment.go:1482] ["delete segment from memory"] [traceID=a9c6dbe891d0e895a6468f9d3f704995] [collectionID=448960188666872753]       [partitionID=448960188666872762] [segmentID=450044002146795106] [segmentType=Sealed] [insertCount=836176]
25668 [2024/05/31 17:30:14.206 +00:00] [DEBUG] [segments/collection.go:217] ["collection ref decrement"] [collectionID=448960188666872753] [refCount=20]
25669 I20240531 17:30:14.223372   175 DataCodec.cpp:123] [SERVER][DeserializeFileData][milvus] DeserializeFileData_deserialize_duration_ms:105
25670 I20240531 17:30:14.240585   178 DataCodec.cpp:123] [SERVER][DeserializeFileData][milvus] DeserializeFileData_deserialize_duration_ms:98
25671 I20240531 17:30:14.280076   174 DataCodec.cpp:123] [SERVER][DeserializeFileData][milvus] DeserializeFileData_deserialize_duration_ms:127
25672 [2024/05/31 17:30:14.290 +00:00] [INFO] [segments/segment.go:1482] ["delete segment from memory"] [traceID=d980913f12e84efcb4f046ae5ed99a7e] [collectionID=448960188666872753]       [partitionID=448960188666872807] [segmentID=450044002146805531] [segmentType=Sealed] [insertCount=836790]
25673 [2024/05/31 17:30:14.290 +00:00] [DEBUG] [segments/collection.go:217] ["collection ref decrement"] [collectionID=448960188666872753] [refCount=19]
25674 I20240531 17:30:14.312639   169 DataCodec.cpp:123] [SERVER][DeserializeFileData][milvus] DeserializeFileData_deserialize_duration_ms:97
25675 [2024/05/31 17:30:14.312 +00:00] [INFO] [segments/segment.go:1482] ["delete segment from memory"] [traceID=d32a0b4a6eb5e8e7d8c223f79f8bd683] [collectionID=448960188666872753]       [partitionID=448960188666872790] [segmentID=450044002146810067] [segmentType=Sealed] [insertCount=838134]
```
3. coredump file in 4am-node36 /tmp/cores:
```
-rw------- 1 root root 104491024384 May 31 17:30 core-laion1b-test-2-milvus-querynode-0-dd6f7496f-6sfdr-milvus-8-1717176620
```
### argo: [laion1b-test-cron-start-59651800](https://argo-workflows.zilliz.cc/archived-workflows/qa/7c7504cc-5adc-43f9-a345-993e9b04a2ac?nodeId=laion1b-test-cron-start-59651800)
1. concurrent: upsert + search + query + load + flush
![image](https://github.com/milvus-io/milvus/assets/27288593/2bfca859-9d0b-4fe9-aaad-846ffc47ddd9)
2. queryNode `laion1b-test-2-milvus-querynode-0-dd6f7496f-5rvdn` terminated with log:
[qn_5rvdn.log](https://github.com/user-attachments/files/15542435/qn_5rvdn.log)

```
29695 [2024/06/03 06:57:36.589 +00:00] [INFO] [segments/segment.go:310] ["create segment"] [traceID=a644e6e94133e65f2c909c0af933ebb2] [collectionID=448960188666872753] [partitionID=      448960188666872805] [segmentID=450044002156756035] [segmentType=Sealed] [level=L1]
29696 [2024/06/03 06:57:36.589 +00:00] [INFO] [segments/segment_loader.go:727] ["start to load segments in parallel"] [traceID=a644e6e94133e65f2c909c0af933ebb2] [collectionID=448960      188666872753] [segmentType=Sealed] [requestSegments="[450044002156756035]"] [preparedSegments="[450044002156756035]"] [segmentNum=1] [concurrencyLevel=1]
29697 terminate called after throwing an instance of 'milvus::SegcoreError'
29698   what():  Error:RemoveDir:Directory not empty
29699 SIGABRT: abort
29700 PC=0x7f1bfd2c003b m=35382 sigcode=18446744073709551610
29701 signal arrived during cgo execution
29702 
29703 goroutine 55081499 [syscall]:
29704 I20240603 06:57:36.642289   174 DataCodec.cpp:123] [SERVER][DeserializeFileData][milvus] DeserializeFileData_deserialize_duration_ms:77
29705 [2024/06/03 06:57:36.705 +00:00] [INFO] [segments/segment.go:1482] ["delete segment from memory"] [traceID=183dd08f46a5cde1a0d273d90a62e23f] [collectionID=448960188666872753]       [partitionID=448960188666872759] [segmentID=450044002156754743] [segmentType=Sealed] [insertCount=931409]
29706 [2024/06/03 06:57:36.705 +00:00] [DEBUG] [segments/collection.go:217] ["collection ref decrement"] [collectionID=448960188666872753] [refCount=25]
29707 I20240603 06:57:36.778534   161 DataCodec.cpp:123] [SERVER][DeserializeFileData][milvus] DeserializeFileData_deserialize_duration_ms:142
29708 [2024/06/03 06:57:36.787 +00:00] [INFO] [segments/segment.go:1482] ["delete segment from memory"] [traceID=b254b06b349f4df9c77ecb8fd9c41931] [collectionID=448960188666872753]       [partitionID=448960188666872788] [segmentID=450044002156781876] [segmentType=Sealed] [insertCount=930496]
29709 [2024/06/03 06:57:36.787 +00:00] [DEBUG] [segments/collection.go:217] ["collection ref decrement"] [collectionID=448960188666872753] [refCount=24]
29710 I20240603 06:57:36.825220   159 DataCodec.cpp:123] [SERVER][DeserializeFileData][milvus] DeserializeFileData_deserialize_duration_ms:89
29711 I20240603 06:57:36.860358   171 DataCodec.cpp:123] [SERVER][DeserializeFileData][milvus] DeserializeFileData_deserialize_duration_ms:78
29712 I20240603 06:57:36.895104   169 DataCodec.cpp:123] [SERVER][DeserializeFileData][milvus] DeserializeFileData_deserialize_duration_ms:77
29713 I20240603 06:57:36.941850   167 DataCodec.cpp:123] [SERVER][DeserializeFileData][milvus] DeserializeFileData_deserialize_duration_ms:75
29714 I20240603 06:57:37.070715   166 DataCodec.cpp:123] [SERVER][DeserializeFileData][milvus] DeserializeFileData_deserialize_duration_ms:75
29715 I20240603 06:57:37.265547   168 DataCodec.cpp:123] [SERVER][DeserializeFileData][milvus] DeserializeFileData_deserialize_duration_ms:76
29716 I20240603 06:57:37.275707   164 DataCodec.cpp:123] [SERVER][DeserializeFileData][milvus] DeserializeFileData_deserialize_duration_ms:77
29717 I20240603 06:57:37.279984   163 DataCodec.cpp:123] [SERVER][DeserializeFileData][milvus] DeserializeFileData_deserialize_duration_ms:76
29718 non-Go function
29719     pc=0x7f1bfd2c003b
29720 non-Go function
29721     pc=0x7f1bfd29f858
29722 non-Go function
```
3. coredump file in 4am-node28:
```
-rw------- 1 root root 198755774464 Jun  3 06:57 core-laion1b-test-2-milvus-querynode-0-dd6f7496f-5rvdn-milvus-7-1717397862
```

### Milvus Log

pods:
```
laion1b-test-2-etcd-0                                             1/1     Running            1 (144d ago)       171d    10.104.25.31    4am-node30   <none>           <none>
laion1b-test-2-etcd-1                                             1/1     Running            0                  171d    10.104.30.94    4am-node38   <none>           <none>
laion1b-test-2-etcd-2                                             1/1     Running            0                  171d    10.104.34.225   4am-node37   <none>           <none>
laion1b-test-2-milvus-datanode-7d95c68964-d8zqd                   1/1     Running            0                  8d      10.104.13.73    4am-node16   <none>           <none>
laion1b-test-2-milvus-datanode-7d95c68964-k6bl7                   1/1     Running            0                  8d      10.104.16.152   4am-node21   <none>           <none>
laion1b-test-2-milvus-indexnode-795c987f94-2nk76                  1/1     Running            0                  8d      10.104.26.151   4am-node32   <none>           <none>
laion1b-test-2-milvus-indexnode-795c987f94-g296m                  1/1     Running            0                  8d      10.104.17.70    4am-node23   <none>           <none>
laion1b-test-2-milvus-indexnode-795c987f94-vxpsh                  1/1     Running            0                  8d      10.104.23.224   4am-node27   <none>           <none>
laion1b-test-2-milvus-mixcoord-5755449b5c-77gbk                   1/1     Running            0                  8d      10.104.34.103   4am-node37   <none>           <none>
laion1b-test-2-milvus-proxy-6c7f7cf9d4-f6nd6                      1/1     Running            0                  8d      10.104.16.153   4am-node21   <none>           <none>
laion1b-test-2-milvus-querynode-0-dd6f7496f-4sjxm                 1/1     Running            0                  8d      10.104.34.104   4am-node37   <none>           <none>
laion1b-test-2-milvus-querynode-0-dd6f7496f-5rvdn                 1/1     Running            1 (20h ago)        8d      10.104.19.196   4am-node28   <none>           <none>
laion1b-test-2-milvus-querynode-0-dd6f7496f-6sfdr                 1/1     Running            1 (3d10h ago)      8d      10.104.33.22    4am-node36   <none>           <none>
laion1b-test-2-milvus-querynode-0-dd6f7496f-xfv2f                 1/1     Running            0                  8d      10.104.32.39    4am-node39   <none>           <none>
laion1b-test-2-pulsar-bookie-0                                    1/1     Running            0                  171d    10.104.33.107   4am-node36   <none>           <none>
laion1b-test-2-pulsar-bookie-1                                    1/1     Running            0                  41d     10.104.18.164   4am-node25   <none>           <none>
laion1b-test-2-pulsar-bookie-2                                    1/1     Running            0                  171d    10.104.25.32    4am-node30   <none>           <none>
laion1b-test-2-pulsar-broker-0                                    1/1     Running            1 (44d ago)        53d     10.104.1.147    4am-node10   <none>           <none>
laion1b-test-2-pulsar-proxy-0                                     1/1     Running            0                  41d     10.104.32.209   4am-node39   <none>           <none>
laion1b-test-2-pulsar-recovery-0                                  1/1     Running            1 (41d ago)        73d     10.104.31.87    4am-node34   <none>           <none>
laion1b-test-2-pulsar-zookeeper-0                                 1/1     Running            0                  171d    10.104.29.87    4am-node35   <none>           <none>
laion1b-test-2-pulsar-zookeeper-1                                 1/1     Running            0                  53d     10.104.21.196   4am-node24   <none>           <none>
laion1b-test-2-pulsar-zookeeper-2                                 1/1     Running            0                  171d    10.104.34.229   4am-node37   <none>           <none>
```

**Comment by chyezh** [comment]
same with #30857

**Comment by chyezh** [comment]
should be fixed at 2.4.5, please verify it.

**Comment by ThreadDao** [comment]
stale
